File: isSameAfterReversals.dart
// https://leetcode.com/problems/a-number-after-a-double-reversal/

void main() {
  print(isSameAfterReversals(526));
}

isSameAfterReversals(num) {
  reverse(int num) {
    String num1 = (num).toString();
    var rev = "";
    for (var a = num1.length - 1; a >= 0; a--) {
      rev = rev + num1[a];
    }
    return (int.parse(rev));
  }

  ;
  var reversed = reverse(reverse(num));
  return reversed == num;
}

File: isSameAfterReversals.js
// https://leetcode.com/problems/a-number-after-a-double-reversal/

var isSameAfterReversals = function (num) {
  let reverse = (num) => {
    num = String(num);
    let rev = "";
    for (let a = num.length - 1; a >= 0; a--) {
      rev = rev + num[a];
    }
    return Number(rev);
  };
  let reverse1 = reverse(num);
  let reverse2 = reverse(reverse1);
  return reverse2 == num;
};

console.log(isSameAfterReversals(526));
